### PR TITLE
Update config_documentation - Change example of msisdn in allowed_local_3pids

### PR DESCRIPTION
**Pull request by Bitzo**
change the example configuration of msisdn in `allowed_local_3pids`, which is a bit misleading that regex pattern has leading '+'

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] [Code style](https://element-hq.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Thanks for the correction! Looks like `+` shouldn't be included in msisdn's at all, so the example configuration was in fact invalid.

I think it'd be worth linking to https://spec.matrix.org/latest/appendices/#3pid-types in the documentation, so that sysadmins know what type of values to expect. Could you add that to the description of the `allowed_local_3pids` option please?

This looks good to me otherwise.

**Comment by Bitzo** [comment]
> Thanks for the correction! Looks like `+` shouldn't be included in msisdn's at all, so the example configuration was in fact invalid.
> 
> I think it'd be worth linking to https://spec.matrix.org/latest/appendices/#3pid-types in the documentation, so that sysadmins know what type of values to expect. Could you add that to the description of the `allowed_local_3pids` option please?
> 
> This looks good to me otherwise.

OK, I have added that.

File: changelog.d/17476.doc [hunk 1]
```diff
@@ -0,0 +1 @@
+Change the example configuration of msisdn in allowed_local_3pids, which is a bit misleading that regex pattern has leading '+'.
```
Comment (anoadragon453) [inline changelog.d/17476.doc:1]:
```suggestion
Update the [`allowed_local_3pids`](https://element-hq.github.io/synapse/v1.112/usage/configuration/config_documentation.html#allowed_local_3pids) config option's msisdn address to a working example.
```

File: docs/usage/configuration/config_documentation.md [hunk 2]
```diff
@@ -2412,6 +2412,8 @@ disable_msisdn_registration: true
 Mandate that users are only allowed to associate certain formats of
 3PIDs with accounts on this server, as specified by the `medium` and `pattern` sub-options.
 
+You can get more information about 3PID and the expected types of `medium` [here](https://spec.matrix.org/latest/appendices/#3pid-types).
```
Comment (anoadragon453) [inline docs/usage/configuration/config_documentation.md:2415]:
```suggestion
More information about 3PIDs, allowed `medium` types and their `address` syntax can be found [in the Matrix spec](https://spec.matrix.org/latest/appendices/#3pid-types).
```

File: docs/usage/configuration/config_documentation.md [hunk 3]
```diff
@@ -2412,6 +2412,8 @@ disable_msisdn_registration: true
 Mandate that users are only allowed to associate certain formats of
 3PIDs with accounts on this server, as specified by the `medium` and `pattern` sub-options.
```
Comment (anoadragon453) [inline docs/usage/configuration/config_documentation.md:2413]:
```suggestion
3PIDs with accounts on this server, as specified by the `medium` and `pattern` sub-options.
`pattern` is a [Perl-like regular expression](https://docs.python.org/3/library/re.html#module-re).
```

**Review by anoadragon453 — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**